File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    let stack = [];
    let flag = false;
    
    s.split("").forEach((e) => {
        if(e === "(") stack.push("(");
        else if(e === ")"){
            if(stack.length === 0){
                flag = true;
            }
            else stack.pop();
        };
    });

    if(flag) return false;
    return stack.length === 0;
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    stack = []

    
    for str in s:
        if str == "(":
            stack.append(str)
        else:
            if len(stack) == 0:
                return False
            else:
                stack.pop()
    
    if len(stack):
        return False
    else:
        return True
    
    